### DISCUSSION TITLE
Extracting shadows as a geometry

### DISCUSSION THREAD

**Discussion by Saeid-Emam** [iTwin/itwinjs-core#2740] [Q&A]
Is there any way to extract shadows as a geometry (like polygon). I want to do so for a specific building not the whole model and in a specific time. This geometry class can even be a bounding box or so. To my knowledge, currently the shadows do not have vector type.

Any help would be appreciated.
![Capture](https://user-images.githubusercontent.com/75958036/142469747-b5dc8a20-b2ec-4290-a772-803412b0d012.PNG)

**Comment by pmconne** [comment]
There is currently no way to do that.
What do you intend to do with the polygon?

**Reply by Saeid-Emam** [reply]
Thank you for your reply.
I want to create a volumetric shadow for that building, like the one is shown in the below picture. I was hoping that we could access to the shadow points on the terrain.
![3](https://user-images.githubusercontent.com/75958036/142472849-83fc973d-b576-4871-a0a9-1c1eaa0b0586.JPG)

**Comment by RBBentley** [comment]
Unfortunately there isn't currently a way to do this in iTwinJs -  There is support for shadow extraction in MicroStation however.
